packages/react/src/components/AccountSettings/ChangePassword/ChangePassword.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+
+import { Logger } from 'aws-amplify';
+import { changePassword, translate } from '@aws-amplify/ui';
+
+import { useAuth } from '../../../internal';
+import { View, Flex } from '../../../primitives';
+import {
+  DefaultCurrentPassword,
+  DefaultError,
+  DefaultNewPassword,
+  DefaultSubmitButton,
+} from './defaultComponents';
+import { ChangePasswordProps } from './types';
+import { FormValues } from '../types';
+
+const logger = new Logger('ChangePassword');
+
+const ChangePassword: React.ComponentType<ChangePasswordProps> = (props) => {
+  const { onSuccess, onError } = props;
+  const [error, setError] = React.useState<string>(null);
+  const [formValues, setFormValues] = React.useState<FormValues>({});
+
+  const { user, isLoading: isAuthUserLoading } = useAuth();
+
+  /** Translations */
+  // TODO: add AccountSettingsTextUtil to collect these strings
+  const currentPasswordLabel = translate('Current Password');
+  const newPasswordLabel = translate('New Password');
+  const updatePasswordText = translate('Update password');
+
+  /** Auth API Handler */
+  const handleChangePassword = React.useCallback(async () => {
+    const { currentPassword, newPassword } = formValues;
+    try {
+      await changePassword({ user, currentPassword, newPassword });
+      setError(null);
```

Comment:
Should we move this above the `await` because it will block until it returns right? If there was a previous error should we hide the error if there is a previous error while we wait for a response?

Reply:
That's a fair point, adjusting.

Reply:
Agree with @dbanksdesign comment. Additionally would only call this if `error` is not `null`

Reply:
Done in [05472b6](https://github.com/aws-amplify/amplify-ui/pull/2885/commits/05472b6582c90f19b53f6165062450415fb1a779) 🙏